### ISSUE TITLE
Add JSON analog of XMLSchedulingDataProcessorPlugin

### ISSUE THREAD

**Issue by nazarovsa** [quartznet/quartznet#1755]
**Is your feature request related to a problem? Please describe.**
There is no problem as such. But I would gladly if there was a mechanism to configure jobs from JSON.

**Describe the solution you'd like**
I'd like to have an opportunity to configure jobs from JSON files like XML. Generally, my suggestion is to add an implementation of ISchedulerPlugin (IFileScanListener possible too) which will run jobs from JSON like a XMLSchedulingDataProcessorPlugin.

**Additional context**
I'm understanding, that might be kind of overengineering because we already have XMLSchedulingDataProcessorPlugin and it works fine. But I have some free time and will be glad to implement it. I asking for an opportunity here, because I didn't find a contribution guide.

**Comment by lahma** [comment]
Thanks for the idea. I'm honestly a bit on the fence about this. Some thoughts:

**PROS**

* JSON is the way to things nowadays, though lacking proper schema support
* would allow appsettings.json integration probably for small setups, could read from there too (quartz:scheduling)
* more ".NET Core like"

**CONS**

* Like you've described there's already XML support and intellise-sense kind-of-experience with the schema validation
* Another thing to maintain, does not give much advantage over XML (?), would need the same tests etc

What's your main thing guiding towards JSON instead of XML (expect for syntax)?

**Comment by nazarovsa** [comment]
Yeah. You are absolutely right about pros, and it's main reasons to implement that feature. Additionally json files requires less space on disk, it can be good for projects with big amount of jobs in file. But this reason is insignificant in terms of saving space.

> JSON is the way to things nowadays, though lacking proper schema support
I think it's not 100% true about schema. Json Schema gives us options for validation, for instance [NJsonSchema](https://github.com/RicoSuter/NJsonSchema).

> What's your main thing guiding towards JSON instead of XML (except for syntax)?
In my opinion, syntax is main reason here.

**Comment by lahma** [comment]
Implemented in #3012 (3.x) and #3014 (4.x port). Adds hierarchical JSON configuration in appsettings.json, declarative job/trigger definitions via a Schedule section, standalone quartz_jobs.json file support via JsonSchedulingDataProcessorPlugin (with hot-reload), and auto-discovery of named schedulers via a Schedulers section.
